src/otg_fs.rs

```diff
@@ -0,0 +1,77 @@
+//! USB OTG full-speed peripheral
+//!
+//! Requires the `usb_fs` feature.
+//! Only one of the `usb_fs`/`usb_hs` features can be selected at the same time.
+
+use crate::pac;
+
+use crate::gpio::{
+    gpioa::{PA11, PA12},
+    Alternate, AF10,
+};
+use crate::rcc::Clocks;
+use crate::time::Hertz;
+
+pub use synopsys_usb_otg::UsbBus;
+use synopsys_usb_otg::UsbPeripheral;
+
+pub struct USB {
+    pub usb_global: pac::OTG_FS_GLOBAL,
+    pub usb_device: pac::OTG_FS_DEVICE,
+    pub usb_pwrclk: pac::OTG_FS_PWRCLK,
+    pub pin_dm: PA11<Alternate<AF10>>,
+    pub pin_dp: PA12<Alternate<AF10>>,
+    pub hclk: Hertz,
+}
+
+impl USB {
+    /// Construct a USB peripheral wrapper.
+    ///
+    /// Call `UsbBus::new` to construct and initialize the USB peripheral driver.
+    pub fn new(
+        usb_global: pac::OTG_FS_GLOBAL,
+        usb_device: pac::OTG_FS_DEVICE,
+        usb_pwrclk: pac::OTG_FS_PWRCLK,
+        pin_dm: PA11<Alternate<AF10>>,
+        pin_dp: PA12<Alternate<AF10>>,
+        clocks: Clocks,
+    ) -> Self {
```

Comment:
IMO the pins should be specified using a tuple to be consistent with the other peripherals. But I don't think there is any benefit in adding a `Pins` trait, like in the other peripherals, if there is only one valid pair of pins.
```suggestion
    pub fn new(
        usb_global: pac::OTG_FS_GLOBAL,
        usb_device: pac::OTG_FS_DEVICE,
        usb_pwrclk: pac::OTG_FS_PWRCLK,
        pins: (PA11<Alternate<AF10>>, PA12<Alternate<AF10>>),
        clocks: Clocks,
    ) -> Self {
```
or perhaps it's better to define a type for the pins:
```rust
pub type USB_DM = PA11<Alternate<AF10>>;
pub type USB_DP = PA12<Alternate<AF10>>;
// maybe: pub type USB_PINS = (USB_DM, USB_DP);
```

Reply:
I don't like neither of these options as they do not tell which pin is D+ and which is D-. What do you think about adding a type `struct Pins { dp: ..., dm: ...}` and passing it instead of a tuple?

Reply:
I agree that this isn't perfect, but I would prefer to be consistent with the other peripherals (in this and other STM32 HAL crates). The order of RX and TX for `Serial::new` can also be mixed up easily, but the compiler will complain if you get the order wrong.

Reply:
Got it, fixed it :)